src/azure-cli/azure/cli/command_modules/servicefabric/tests/latest/test_sf_application.py

```diff
@@ -89,7 +89,7 @@ def _app_service_test(self):
 
     @ResourceGroupPreparer()
     @KeyVaultPreparer(name_prefix='sfrp-cli-kv-', additional_params='--enabled-for-deployment --enabled-for-template-deployment')
-    def test_application(self, key_vault):
+    def test_application(self, key_vault, resource_group):
```

Comment:
Why you need to add parameter `resource_group` to those test methods? It should be assigned to `{rg}` by default

Reply:
without this I found that the teardown of keyvault would break due to this line https://github.com/Azure/azure-python-devtools/blob/master/src/azure_devtools/scenario_tests/preparers.py#L53. The resource_group kwarg that the resource group preparer provides gets removed by that trim function and then the keyt vault preparer teardown function https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-testsdk/azure/cli/testsdk/preparers.py#L189 which relies on that kwarg instead gets None and then the remove command errors out

Reply:
OK, got it~